apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/ClientCacheExtensions.kt

```diff
@@ -95,14 +99,83 @@ fun ApolloClient.Builder.store(store: ApolloStore, writeToCacheAsynchronously: B
       .writeToCacheAsynchronously(writeToCacheAsynchronously)
 }
 
+enum class WatchErrorHandling {
+  THROW_CACHE_ERRORS,
+  THROW_NETWORK_ERRORS,
+  THROW_CACHE_AND_NETWORK_ERRORS,
+  IGNORE_ERRORS
+}
+
 /***
  * Gets the result from the network, then observes the cache for any changes.
  * Overriding the [FetchPolicy] will change how the result is first queried.
+ * Exception are ignored by default, this can be changed by setting [fetchErrorHandling] for the first fetch and [refetchErrorHandling]
+ * for subsequent fetches.
+ */
+@JvmOverloads
+fun <D : Query.Data> ApolloCall<D>.watch(
+    fetchErrorHandling: WatchErrorHandling = WatchErrorHandling.IGNORE_ERRORS,
+    refetchErrorHandling: WatchErrorHandling = WatchErrorHandling.IGNORE_ERRORS,
```

Comment:
Also, passing `fetchError` and `refetchError` kind of breaks the usual pattern of how `ApolloCalls` are configured. Maybe this instead?

```kotlin
apolloClient.query(query)
             .fetchPolicy(CacheFirst)
             .refetchPolicy(CacheOnly)
             .errorHandling(THROW_CACHE_AND_NETWORK_ERRORS)
             .refetchErrorHandling(IGNORE_ERROR)
             .watch()
```

2 advantages:

1. It's more consistent
2. It allows to clone the call and keep all parameters (not sure if that is ever going to be used but maybe? )

Reply:
Agreed! 👍

I'll also make them CamelCase (we use both conventions in the codebase, but there's `FetchPolicy` right above it which uses CamelCase so it will be at least consistent in the file :)

Reply:
Yay for CamelCase 🙌 

Also maybe `ThrowAll` instead of the verbose `ThrowCacheAndNetworkErrors` ?

Reply:
Question: did you mean that `errorHandling` should be taken into account also outside of `watch`? If yes it may be slightly confusing as the default is to ignore in `watch` but throw all in `toFlow`/`execute` - but this can be clarified in the KDocs.

Reply:
> did you mean that errorHandling should be taken into account also outside of watch

I'm not sure... On one hand, it being a general thing makes it more consistent. On the other hand, it doesn't bring much over using `.catch {}` directly so it's adding noise in the API. I think I'd limit this to `watch()` for the time being?